### PR TITLE
fix check warnings/notes in benchmark package

### PR DESCRIPTION
**Pull request by moki1202**

### CODE REVIEW

File: modules/benchmark/R/align_data.R [hunk 1]
```diff
@@ -43,8 +43,8 @@ align_data <- function(model.calc, obvs.calc, var, align_method = "match_timeste
   coarse.unit <- compare$diff_units[coarse]
   
   # Round to the larger time step (experimental)
-  obvs.calc$round.posix  <- as.POSIXct(round(obvs.calc$posix,  units = coarse.unit))
-  model.calc$round.posix <- as.POSIXct(round(model.calc$posix, units = coarse.unit))
+  obvs.calc$round.posix  <- as.POSIXct(round(obvs.calc$posix),  units = coarse.unit)
+  model.calc$round.posix <- as.POSIXct(round(model.calc$posix), units = coarse.unit)
```
Comment (infotroph) [inline modules/benchmark/R/align_data.R:47]: This is a tricky one, but I think `units` was correctly inside the `round` call here -- the calculation will dispatch to `round.POSIXt`, which accepts a `units` argument even though none is documented for the `round` generic. 
```suggestion
  obvs.calc$round.posix  <- as.POSIXct(round(obvs.calc$posix,  units = coarse.unit))
  model.calc$round.posix <- as.POSIXct(round(model.calc$posix, units = coarse.unit))
```

File: modules/benchmark/R/metric_residual_plot.R [hunk 2]
```diff
@@ -12,17 +12,17 @@ metric_residual_plot <- function(metric_dat, var, filename = NA, draw.plot = is.
...
   metric_dat$diff <- abs(metric_dat$model - metric_dat$obvs)
-  metric_dat$zeros <- rep(0, length(time))
+  metric_dat$zeros <- rep(0, length(metric_dat$time))
   
-  p <- ggplot2::ggplot(data = metric_dat, aes(x = time)) 
-  p <- p + ggplot2::geom_path(aes(y = zeros), colour = "#666666", size = 2, linetype = 2, lineend = "round") 
-  p <- p + ggplot2::geom_point(aes(y = diff), size = 4, colour = "#619CFF") 
+  p <- ggplot2::ggplot(data = metric_dat, ggplot2::aes(x = metric_dat$time)) 
+  p <- p + ggplot2::geom_path(ggplot2::aes(y = metric_dat$zeros), colour = "#666666", size = 2, linetype = 2, lineend = "round") 
+  p <- p + ggplot2::geom_point(ggplot2::aes(y = diff), size = 4, colour = "#619CFF")
```
Comment (infotroph) [inline modules/benchmark/R/metric_residual_plot.R:19]:
```suggestion
    p <- ggplot2::ggplot(data = metric_dat, ggplot2::aes(x = .data$time)) 
  p <- p + ggplot2::geom_path(ggplot2::aes(y = .data$zeros), colour = "#666666", size = 2, linetype = 2, lineend = "round") 
  p <- p + ggplot2::geom_point(ggplot2::aes(y = .data$diff), size = 4, colour = "#619CFF") 
```

File: modules/benchmark/R/metric_timeseries_plot.R [hunk 3]
```diff
@@ -19,17 +19,17 @@ metric_timeseries_plot <- function(metric_dat, var, filename = NA, draw.plot = i
     metric_dat$time <- date.time
   }
   
-  p <- ggplot(data = metric_dat, aes(x = time)) 
+  p <- ggplot(data = metric_dat, ggplot2::aes(x = metric_dat$time))
```
Comment (infotroph) [inline modules/benchmark/R/metric_timeseries_plot.R:22]:
```suggestion
  p <- ggplot(data = metric_dat, ggplot2::aes(x = .data$time)) 
```

File: modules/benchmark/R/metric_lmDiag_plot.R [hunk 4]
```diff
@@ -7,9 +7,9 @@
 metric_lmDiag_plot <- function(metric_dat, var, filename = NA, draw.plot = FALSE) {
   PEcAn.logger::logger.info("Metric: Linear Regression Diagnostic Plot")
   
-  fit <- lm(metric_dat[, 1] ~ metric_dat[, 2])
+  fit <- stats::lm(metric_dat[, 1] ~ metric_dat[, 2])
 
-  p1 <- ggplot2::ggplot(fit, aes(.fitted, .resid))
+  p1 <- ggplot2::ggplot(fit, ggplot2::aes(.fitted, .resid))
```
Comment (infotroph) [inline modules/benchmark/R/metric_lmDiag_plot.R:12]:
```suggestion
  p1 <- ggplot2::ggplot(fit, ggplot2::aes(.data$.fitted, .data$.resid))
```
Here's another sneaky one: Where did `.fitted` and `.resid` come from and how do we keep R CMD check from complaining about them? Well! 
Screenshot 2024-02-13 at 1 34 06 PM
It turns out ggplot is converting the `fit` object into a data frame under the hood using `fortify()`, so we can pass any of the names `fortify.lm` creates into `.data` just as if they were already present in the dataframe.
Screenshot 2024-02-13 at 1 35 10 PM

Comment (infotroph) [reply modules/benchmark/R/metric_lmDiag_plot.R:12]: (This feels like a lot of magic to rely on here, but when trying to make the package checks quiet down I think simply prefixing `.data$` on what's already here is the right call)

File: modules/benchmark/R/metric_lmDiag_plot.R [hunk 5]
```diff
@@ -18,31 +18,31 @@ metric_lmDiag_plot <- function(metric_dat, var, filename = NA, draw.plot = FALSE
   p1 <- p1 + ggplot2::ggtitle("Residual vs Fitted Plot") 
   p1 <- p1 + ggplot2::theme_bw()
   
-  p2 <- ggplot2::ggplot(fit, aes(qqnorm(.stdresid)[[1]], .stdresid)) 
+  p2 <- ggplot2::ggplot(fit, ggplot2::aes(stats::qqnorm(.stdresid)[[1]], .stdresid))
```
Comment (infotroph) [inline modules/benchmark/R/metric_lmDiag_plot.R:21]:
```suggestion
  p2 <- ggplot2::ggplot(fit, ggplot2::aes(stats::qqnorm(.data$.stdresid)[[1]], .data$.stdresid)) 
```

File: modules/benchmark/R/metric_lmDiag_plot.R [hunk 6]
```diff
@@ -18,31 +18,31 @@ metric_lmDiag_plot <- function(metric_dat, var, filename = NA, draw.plot = FALSE
   p1 <- p1 + ggplot2::ggtitle("Residual vs Fitted Plot") 
   p1 <- p1 + ggplot2::theme_bw()
   
-  p2 <- ggplot2::ggplot(fit, aes(qqnorm(.stdresid)[[1]], .stdresid)) 
+  p2 <- ggplot2::ggplot(fit, ggplot2::aes(stats::qqnorm(.stdresid)[[1]], .stdresid)) 
   p2 <- p2 + ggplot2::geom_point(na.rm = TRUE)
-  p2 <- p2 + ggplot2::geom_abline(aes(qqline(.stdresid))) 
+  p2 <- p2 + ggplot2::geom_abline(ggplot2::aes(stats::qqline(.stdresid)))
```
Comment (infotroph) [inline modules/benchmark/R/metric_lmDiag_plot.R:23]:
```suggestion
  p2 <- p2 + ggplot2::geom_abline(ggplot2::aes(stats::qqline(.data$.stdresid))) 
```

File: modules/benchmark/R/metric_lmDiag_plot.R [hunk 7]
```diff
@@ -18,31 +18,31 @@ metric_lmDiag_plot <- function(metric_dat, var, filename = NA, draw.plot = FALSE
...
   p2 <- p2 + ggplot2::geom_point(na.rm = TRUE)
-  p2 <- p2 + ggplot2::geom_abline(aes(qqline(.stdresid))) 
+  p2 <- p2 + ggplot2::geom_abline(ggplot2::aes(stats::qqline(.stdresid))) 
   p2 <- p2 + ggplot2::xlab("Theoretical Quantiles") 
   p2 <- p2 + ggplot2::ylab("Standardized Residuals")
   p2 <- p2 + ggplot2::ggtitle("Normal Q-Q") 
   p2 <- p2 + ggplot2::theme_bw()
   
-  p3 <- ggplot2::ggplot(fit, aes(.fitted, sqrt(abs(.stdresid)))) 
+  p3 <- ggplot2::ggplot(fit, ggplot2::aes(.fitted, sqrt(abs(.stdresid))))
```
Comment (infotroph) [inline modules/benchmark/R/metric_lmDiag_plot.R:29]:
```suggestion
  p3 <- ggplot2::ggplot(fit, ggplot2::aes(.fitted, sqrt(abs(.data$.stdresid)))) 
```

File: modules/benchmark/R/metric_lmDiag_plot.R [hunk 8]
```diff
@@ -18,31 +18,31 @@ metric_lmDiag_plot <- function(metric_dat, var, filename = NA, draw.plot = FALSE
...
+  p3 <- ggplot2::ggplot(fit, ggplot2::aes(.fitted, sqrt(abs(.stdresid)))) 
   p3 <- p3 + ggplot2::geom_point(na.rm = TRUE)
   p3 <- p3 + ggplot2::stat_smooth(method = "loess", na.rm = TRUE) 
   p3 <- p3 + ggplot2::xlab("Fitted Value")
   p3 <- p3 + ggplot2::ylab(expression(sqrt("|Standardized residuals|")))
   p3 <- p3 + ggplot2::ggtitle("Scale-Location") 
   p3 <- p3 + ggplot2::theme_bw()
   
-  p4 <- ggplot2:: ggplot(fit, aes(seq_along(.cooksd), .cooksd)) 
+  p4 <- ggplot2:: ggplot(fit, ggplot2::aes(seq_along(.cooksd), .cooksd))
```
Comment (infotroph) [inline modules/benchmark/R/metric_lmDiag_plot.R:37]:
```suggestion
  p4 <- ggplot2:: ggplot(fit, ggplot2::aes(seq_along(.data$.cooksd), .data$.cooksd)) 
```

File: modules/benchmark/R/metric_lmDiag_plot.R [hunk 9]
```diff
@@ -18,31 +18,31 @@ metric_lmDiag_plot <- function(metric_dat, var, filename = NA, draw.plot = FALSE
...
   p4 <- p4 + ggplot2::geom_bar(stat = "identity", position = "identity")
   p4 <- p4 + ggplot2::xlab("Obs. Number") 
   p4 <- p4 + ggplot2::ylab("Cook's distance")
   p4 <- p4 + ggplot2::ggtitle("Cook's distance") 
   p4 <- p4 + ggplot2::theme_bw()
   
-  p5 <- ggplot2::ggplot(fit, aes(.hat, .stdresid)) 
-  p5 <- p5 + ggplot2::geom_point(aes(size = .cooksd), na.rm = TRUE)
+  p5 <- ggplot2::ggplot(fit, ggplot2::aes(.hat, .stdresid)) 
+  p5 <- p5 + ggplot2::geom_point(ggplot2::aes(size = .cooksd), na.rm = TRUE)
```
Comment (infotroph) [inline modules/benchmark/R/metric_lmDiag_plot.R:45]:
```suggestion
  p5 <- ggplot2::ggplot(fit, ggplot2::aes(.data$.hat, .data$.stdresid)) 
  p5 <- p5 + ggplot2::geom_point(ggplot2::aes(size = .data$.cooksd), na.rm = TRUE)
```

File: modules/benchmark/R/metric_lmDiag_plot.R [hunk 10]
```diff
@@ -51,7 +51,7 @@ metric_lmDiag_plot <- function(metric_dat, var, filename = NA, draw.plot = FALSE
   p5 <- p5 + ggplot2::theme_bw() 
   p5 <- p5 + ggplot2::theme(legend.position = "bottom")
   
-  p6 <- ggplot2::ggplot(fit, aes(.hat, .cooksd)) 
+  p6 <- ggplot2::ggplot(fit, ggplot2::aes(.hat, .cooksd))
```
Comment (infotroph) [inline modules/benchmark/R/metric_lmDiag_plot.R:54]:
```suggestion
  p6 <- ggplot2::ggplot(fit, ggplot2::aes(.data$.hat, .data$.cooksd)) 
```

File: modules/benchmark/R/metric_scatter_plot.R [hunk 11]
```diff
@@ -12,14 +12,14 @@ metric_scatter_plot <- function(metric_dat, var, filename = NA, draw.plot = is.n
   PEcAn.logger::logger.info("Metric: Scatter Plot")
   
   p <- ggplot2::ggplot(data = metric_dat) 
-  p <- p + ggplot2::geom_point(aes(x = model, y = obvs), size = 4) 
+  p <- p + ggplot2::geom_point(ggplot2::aes(x = model, y = obvs), size = 4)
```
Comment (infotroph) [inline modules/benchmark/R/metric_scatter_plot.R:15]:
```suggestion
  p <- p + ggplot2::geom_point(ggplot2::aes(x = .data$model, y = .data$obvs), size = 4) 
```

File: modules/benchmark/R/metric_timeseries_plot.R [hunk 12]
```diff
@@ -19,17 +19,17 @@ metric_timeseries_plot <- function(metric_dat, var, filename = NA, draw.plot = i
...
+  p <- ggplot(data = metric_dat, ggplot2::aes(x = metric_dat$time)) 
   p <- p + labs(title = var, y = "") 
-  p <- p + geom_path(aes(y = model, colour = "Model"), size = 2) 
-  p <- p + geom_point(aes(y = model, colour = "Model"), size = 4) 
-  p <- p + geom_path(aes(y = obvs, colour = "Observed"), size = 2) 
-  p <- p + geom_point(aes(y = obvs, colour = "Observed"), size = 4)
+  p <- p + geom_path(ggplot2::aes(y = model, colour = "Model"), size = 2) 
+  p <- p + geom_point(ggplot2::aes(y = model, colour = "Model"), size = 4) 
+  p <- p + geom_path(ggplot2::aes(y = obvs, colour = "Observed"), size = 2) 
+  p <- p + geom_point(ggplot2::aes(y = obvs, colour = "Observed"), size = 4)
```
Comment (infotroph) [inline modules/benchmark/R/metric_timeseries_plot.R:27]:
```suggestion
  p <- p + geom_path(ggplot2::aes(y = .data$model, colour = "Model"), size = 2) 
  p <- p + geom_point(ggplot2::aes(y = .data$model, colour = "Model"), size = 4) 
  p <- p + geom_path(ggplot2::aes(y = .data$obvs, colour = "Observed"), size = 2) 
  p <- p + geom_point(ggplot2::aes(y = .data$obvs, colour = "Observed"), size = 4)
```

**Review by infotroph — CHANGES_REQUESTED** [review]

File: modules/benchmark/R/metric_residual_plot.R [hunk 13]
```diff
@@ -12,17 +12,17 @@ metric_residual_plot <- function(metric_dat, var, filename = NA, draw.plot = is.
   
   metric_dat$time <- lubridate::year(as.Date(as.character(metric_dat$time), format = "%Y"))
   metric_dat$diff <- abs(metric_dat$model - metric_dat$obvs)
-  metric_dat$zeros <- rep(0, length(time))
+  metric_dat$zeros <- rep(0, length(metric_dat$time))
   
-  p <- ggplot2::ggplot(data = metric_dat, aes(x = time)) 
-  p <- p + ggplot2::geom_path(aes(y = zeros), colour = "#666666", size = 2, linetype = 2, lineend = "round") 
-  p <- p + ggplot2::geom_point(aes(y = diff), size = 4, colour = "#619CFF") 
+    p <- ggplot2::ggplot(data = metric_dat, ggplot2::aes(x = .data$time))
```
Comment (infotroph) [inline modules/benchmark/R/metric_residual_plot.R:17]:
```suggestion
  p <- ggplot2::ggplot(data = metric_dat, ggplot2::aes(x = .data$time)) 
```

**Review by infotroph — APPROVED** [review]

**Review by mdietze — APPROVED** [review]

**Pull request merged**